src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1131,6 +1131,21 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::do_init_tm_tx(
     }
     auto term = term_opt.value();
 
+    auto cfg = _metadata_cache.local().get_topic_cfg(model::tx_manager_nt);
```

Comment:
the initial creation of hosted transactions should happen on the attempt to so local routing

Reply:
I have checked
I think that initialization of hosted transaction should happen in tm initialization.
We are already on leader node, we do it at stm start